### PR TITLE
[Payment Method Improvements] Add more tests

### LINKED ISSUE #11542: [Payment] Testing: Add implementation so order details load successfully, emits success state and all tests pass
Complete adding test classes and expand test cases. Make sure code coverage is increased, not decreased. See this comment:
https://github.com/woocommerce/woocommerce-android/pull/11532#issuecomment-2118303080

```[tasklist]
### Tasks
- [x] Fix unit tests
- [x] Create test plan for QA team: Ship with high quality.See iOS test plan
- [x] Make sure ChangeDueCalculatorViewModelTest has a case for all the fields in the UiState.Success class
```

### LINKED ISSUE #11623: [Payment Method Improvements] Enable FF: OTHER_PAYMENT_METHODS -> true

### PR DESCRIPTION
**Pull request by backwardstruck**
Closes: #11542

This PR is branched off of https://github.com/woocommerce/woocommerce-android/pull/11606

### Description

Add more unit tests to increase test coverage.

### Testing instructions

**Unit tests should pass**

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

---

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by backwardstruck** [comment]
Working on resolving the conflicts with the target branch...

**Review by kidinov — APPROVED** [review]
@backwardstruck 👋

Could you please test that all once again as the changes I was doing a bit in a rush as I am quite overwhelmed with the pos? 

Then you can just change the feature flag. But please do not forget to remove feature flag, `OtherPaymentMethodsEnabled` class and all the "old" cash collection code - dialog/tests and so on in the following PR (can be merged to the next release)

**Comment by backwardstruck** [comment]
> @backwardstruck 👋
> 
> Could you please test that all once again as the changes I was doing a bit in a rush as I am quite overwhelmed with the pos?
> 
> Then you can just change the feature flag. But please do not forget to remove feature flag, `OtherPaymentMethodsEnabled` class and all the "old" cash collection code - dialog/tests and so on in the following PR (can be merged to the next release)

Sure! I created this ticket for that code removal:
https://github.com/woocommerce/woocommerce-android/issues/11625

Testing now. And changing the FF value to `true`

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/217">18.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt [hunk 1]
```diff
@@ -26,15 +26,14 @@ enum class FeatureFlag {
...
-            OTHER_PAYMENT_METHODS -> false
-
             MORE_MENU_INBOX,
             WOO_POS,
             WC_SHIPPING_BANNER,
             BETTER_CUSTOMER_SEARCH_M2,
             ORDER_CREATION_AUTO_TAX_RATE,
             DYNAMIC_DASHBOARD_M2 -> PackageUtils.isDebugBuild()
 
+            OTHER_PAYMENT_METHODS -> false
```
Comment (backwardstruck) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/FeatureFlag.kt]: Change this back to `true` before merging.

**Pull request merged**